src/map/skill.cpp

```diff
@@ -3870,6 +3865,21 @@ int64 skill_attack (int attack_type, struct block_list* src, struct block_list *
 			else
 				dmg.dmotion = clif_skill_damage(dsrc, bl, tick, dmg.amotion, dmg.dmotion, damage, dmg.div_, skill_id, skill_lv, dmg_type);
 			break;
+		case AL_INCAGI:
+		case AL_BLESSING:
+		case CASH_BLESSING:
+		case CASH_INCAGI:
+		case MER_BLESSING:
+		case MER_INCAGI:
+			if (tsc && tsc->data[SC_CHANGEUNDEAD]) {
+				damage = 1;
+				dmg.dmotion = clif_skill_damage(dsrc, bl, tick, dmg.amotion, dmg.dmotion, damage, dmg.div_, skill_id, skill_lv, dmg_type);
+
+				if (tstatus->hp == 1)
+					return 0; // No actual damage is given if the target has 1 HP.
+			} else
+				dmg.dmotion = clif_skill_damage(dsrc, bl, tick, dmg.amotion, dmg.dmotion, damage, dmg.div_, skill_id, skill_lv, dmg_type);
```

Comment:
Do we need this if / else? I thought we only call skill_attack when the target has SC_CHANGEUNDEAD in the first place.